Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/BaseTemplateBean.java

```diff
@@ -32,60 +35,95 @@ public abstract class BaseTemplateBean extends BaseIndexedBean {
      */
     public String getTitle() {
         return this.title;
     }
 
     /**
-     * Set title.
+     * Sets the process or process template name. Since the process name is used
+     * in file paths, it should only contain characters compatible with the
+     * operating file system. Also, for scripting, there should be no spaces in
+     * the process name.
      *
-     * @param title as String
+     * @param title
+     *            the process or process template name
      */
     public void setTitle(String title) {
         this.title = title.trim();
     }
 
     /**
-     * Get sortHelperStatus.
-     *
-     * @return value of sortHelperStatus
+     * Returns a coded overview of the progress of the process. The larger the
+     * number, the more advanced the process is, so it can be used to sort by
+     * progress. The numeric code consists of twelve digits, each three digits
+     * from 000 to 100 indicate the percentage of tasks completed, currently in
+     * progress, ready to start and not yet ready, in that order. For example,
+     * 000000025075 means that 25% of the tasks are ready to be started and 75%
+     * of the tasks are not yet ready to be started because previous tasks have
+     * not yet been processed.
+     * 
+     * @return overview of the processing status
      */
     public String getSortHelperStatus() {
         return this.sortHelperStatus;
     }
 
     /**
-     * Set sortHelperStatus.
-     *
-     * @param sortHelperStatus as java.lang.String
+     * Sets the coded overview of the processing status of the process. This
+     * should only be set manually if this information comes from a third-party
+     * source. Typically, sorting progress is determined from the progress
+     * properties of the tasks in the process. The numeric code consists of
+     * twelve digits, each three digits from 000 to 100 indicate the percentage
+     * of tasks completed, currently in progress, ready to start and not yet
+     * ready, in that order. The sum of the four groups of numbers must be 100.
+     * 
+     * @param sortHelperStatus
+     *            coded overview of the progress with pattern
+     *            <code>([01]\d{2}){4}</code>
      */
     public void setSortHelperStatus(String sortHelperStatus) {
         this.sortHelperStatus = sortHelperStatus;
     }
 
     /**
-     * Get creationDate.
+     * Returns the time the process or process template was created.
+     * {@link Date} is a specific instant in time, with millisecond precision.
      *
-     * @return value of creationDate
+     * @return the creation time
      */
     public Date getCreationDate() {
         return this.creationDate;
     }
 
     /**
-     * Set creationDate.
+     * Returns the time the process was created. The string is formatted
+     * according to {@link SimpleDateFormat}{@code ("yyyy-MM-dd HH:mm:ss")}.
+     *
+     * @return the creation time
+     * @deprecated Use {@link #getCreationDate()}.
```

Comment:
Why adding a new method with the `deprecated` annotation? Why not change to the right or to be use method?

Reply:
This is a plumbing method used for objects that previously accessed DTO objects that provided creation time as string, which is not available in the bean. It is marked `@Deprecated` so that over time we can adjust all the places where this functionality is used. I have not made these adjustments yet so that the development will be finished at some point.

Reply:
Understandable too but I have the maybe wrong impression that only methods marked as deprecated but never get removed.

Reply:
I took your feedback as an opportunity to take another look at it yesterday, and was able to remove quite a bit of dead code.

Reply:
I took your feedback as an opportunity to take another look at it yesterday, and was able to remove quite a bit of dead code.